examples/ae_examples/vae_example/client.py

```diff
@@ -0,0 +1,64 @@
+import argparse
+from pathlib import Path
+from typing import Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+from flwr.common.typing import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from examples.ae_examples.vae_example.models import MnistVariationalDecoder, MnistVariationalEncoder
+from fl4health.clients.basic_client import BasicClient
+from fl4health.model_bases.autoencoders_base import AutoEncoderType, VariationalAE
+from fl4health.preprocessing.autoencoders.loss import VAE_loss
+from fl4health.preprocessing.autoencoders.vae_training import AETransformer
+from fl4health.utils.load_data import load_mnist_data
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+class VAEClient(BasicClient):
```

Comment:
We made some code structure assumptions so that we could easily use other FL strategies for training AEs without too many changes. Maybe we can have this example inherit from a more complex client, say FedProx?